apollo-api/src/main/java/com/apollostack/api/Query.java

```diff
@@ -1,17 +1,34 @@
 package com.apollostack.api;
 
-import java.util.List;
+import java.util.Collections;
 import java.util.Map;
 
 /** TODO */
-public interface Query {
+public interface Query<V extends Query.Variables> {
```

Comment:
I still think we don't need this `V` generic type on `Query`. 
First because you don't always have variable, so a lot of times it's just `Query<Query.Variables>` which is not super helpful. 
Second, I think we can just roll with `Query.Variables` all the time, right? Why does the `variables()` method need need to be generic?